homeassistant/components/mqtt/__init__.py

```diff
@@ -585,6 +584,51 @@ class Message:
 class MQTT:
     """Home Assistant MQTT client."""
 
+    class AsyncioHelper:
+        """Helper to handle paho-mqtt async callbacks."""
+
+        def __init__(self, hass, client):
+            """Initialize helper."""
+            self.hass = hass
+            self.task_misc = None
+            self.client = client
+            self.client.on_socket_open = self.on_socket_open
+            self.client.on_socket_close = self.on_socket_close
+            self.client.on_socket_register_write = self.on_socket_reg_write
+            self.client.on_socket_unregister_write = self.on_socket_unreg_write
+
+        def on_socket_open(self, client, userdata, sock):
+            """Socket open callback."""
+            def cb():
+                client.loop_read()
+
+            self.hass.loop.add_reader(sock, cb)
```

Comment:
Is this OK?

Reply:
I think so. It's as in the paho example.